### PR TITLE
feat: add products list page

### PR DESCRIPTION
**Pull request by Edyta2801**

### CODE REVIEW

**Review by dblach-intive — CHANGES_REQUESTED** [review]
Looks good but, we have already our custom Table in: https://github.com/intive/patronage22_lublin_js_cms/tree/main/src/components/Table

Could you use our table because we will be using this one i all tables in our app

**Comment by dblach-intive** [comment]
Please keep commits name and prs convention - you can find info in tracker

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "axios": "^0.25.0",
     "formik": "^2.2.9",
     "framer-motion": "^4.1.17",
+    "moment": "^2.29.1",
```
Comment (patrykomiotek) [inline package.json:23]: moment is no longer developed anymore. We can use https://date-fns.org/ instead

File: src/components/Layout/index.tsx [hunk 2]
```diff
@@ -48,7 +49,7 @@ const Layout = () => {
         <Toolbar />
         <Switch>
           <Route exact path="/dashboard" render={() => <Dashboard />} />
-        </Switch>
+          <Route exact path="/products" render={() => <Products />} /></Switch>
```
Comment (patrykomiotek) [inline src/components/Layout/index.tsx:52]: it's render props pattern and is not needed here. We can write:

```jsx
<Route exact path="/products" /><Products /></Route/>
```

or 

```jsx
<Route exact path="/products" component={Products} /></Route/>
```

**Review by dblach-intive — CHANGES_REQUESTED** [review]
Zrzut ekranu 2022-03-1 o 10 14 09

File: src/components/Layout/index.tsx [hunk 3]
```diff
@@ -48,7 +49,8 @@ const Layout = () => {
         <Toolbar />
         <Switch>
           <Route exact path="/dashboard" render={() => <Dashboard />} />
-        </Switch>
+          <Route exact path="/products" /> <Products /><Route/>
```
Comment (dblach-intive) [inline src/components/Layout/index.tsx:52]: can we use the same as in other places ? ->  <Route exact path={....} component={....} />

File: src/components/Table/EditButton.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import { Button } from "@mui/material";
+
+interface Props {
+    children: React.ReactNode;
+    onClick: () => void;
+  }
+
+const EditButton: React.FC<Props> = ({ children, onClick }) => {
```
Comment (dblach-intive) [inline src/components/Table/EditButton.tsx:9]: we need to pass ID product here

File: src/components/Table/EditButton.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  return (
+    <Button variant="text" color="primary" size="small" onClick={onClick} href={`/products/ID_PRODUCT`}>
```
Comment (dblach-intive) [inline src/components/Table/EditButton.tsx:11]: Id product props should be added not ID_PRODUCT

File: src/components/Table/ProductsRow.tsx [hunk 6]
```diff
@@ -18,14 +21,17 @@ const ProductsRow: React.FC<CustomRowProps> = ({ product }) => {
...
       <StyledTableCell align="center">
         {published ? (
           <Chip clickable label="Yes" color="success" variant="filled" />
         ) : (
           <Chip clickable label="No" color="error" variant="filled" />
         )}
       </StyledTableCell>
+      <StyledTableCell align="center">{`${format(parseISO(createdAt), 'MM/dd/yyyy')}`}</StyledTableCell>
+      <StyledTableCell align="center">
+        <EditButton onClick={() => console.log('Button clicked')}>Edit</EditButton>
```
Comment (dblach-intive) [inline src/components/Table/ProductsRow.tsx:33]: Create new route for edited product and redirect it on onCLick -> "product/edit/Id_produktu

**Review by dblach-intive — CHANGES_REQUESTED** [review]

File: src/components/Table/EditButton.tsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+interface Props {
+    children: React.ReactNode;
+    onClick: () => void;
+    id: number;
+  }
+
+const EditButton: React.FC<Props> = ({ children, onClick }) => {
+  return (
+    <Button variant="text" color="primary" size="small" onClick={onClick}
```
Comment (dblach-intive) [inline src/components/Table/EditButton.tsx:12]: here you can use onClick={() => history.push(`/product/edit/${productId}`)}

File: src/components/Table/EditButton.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import { Button } from "@mui/material";
+
+interface Props {
+    children: React.ReactNode;
+    onClick: () => void;
+    id: number;
+  }
+
+const EditButton: React.FC<Props> = ({ children, onClick }) => {
```
Comment (dblach-intive) [inline src/components/Table/EditButton.tsx:10]: prop productId here or id its up to u

File: src/components/Table/ProductsRow.tsx [hunk 9]
```diff
@@ -2,9 +2,19 @@ import { Chip } from '@mui/material';
 import React from 'react';
 import { CustomRowProps } from '../../types/table';
 import { StyledTableCell, StyledTableRow } from './styles';
+import EditButton from './EditButton';
+import { parseISO, format} from 'date-fns';
+import { useParams, useHistory } from 'react-router-dom';
+
 
 const ProductsRow: React.FC<CustomRowProps> = ({ product }) => {
```
Comment (dblach-intive) [inline src/components/Table/ProductsRow.tsx:10]: you can destructure here {id, title, price, published, createdAt}

File: src/components/Table/ProductsRow.tsx [hunk 10]
```diff
@@ -2,9 +2,19 @@ import { Chip } from '@mui/material';
...
-  const { id, title, price, description, published } = product;
+  const { id, title, price, published, createdAt } = product;
+
+  const history = useHistory();
+  
+let {productId} = useParams<{productId?:any}>();
```
Comment (dblach-intive) [inline src/components/Table/ProductsRow.tsx:15]: you dont have to useParams you have prop with product id

**Review by dblach-intive — CHANGES_REQUESTED** [review]

**Review by dblach-intive — CHANGES_REQUESTED** [review]
you need to add new route product/edit/id - id its a id of product

File: src/components/Table/EditButton.tsx [hunk 11]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
```
Comment (dblach-intive) [inline src/components/Table/EditButton.tsx:1]: Can we move EditButton to components ? not inside table, we will use it in other places too, create components/EditButton catalogue with index.tsx

**Review by Edyta2801 — COMMENTED** [review]
fix

**Review by dblach-intive — APPROVED** [review]

**Pull request merged**